service/controller/v15/resource/drainer/create.go

```diff
@@ -0,0 +1,183 @@
+package drainer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	corev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/aws-operator/service/controller/v15/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/v15/key"
+)
+
+// EnsureCreated tries to drain guest cluster nodes when necessary. Once it
+// detects a guest cluster node being in terminating/wait state in EC2 a
+// DrainerConfig is created to instruct the node-operator to drain the specific
+// node. The node-operator updates the DrainerConfig state as soon as it has
+// drained the node and the aws-operator here completes the lifecycle hook of
+// the drained node and deletes the DrainerConfig.
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	controllerCtx, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	workerASGName := controllerCtx.Status.Drainer.WorkerASGName
+	if workerASGName == "" {
+		// TODO add log fmt entry about controllercontext.Context.Status values retrival.
```

Comment:
The code should never reach that because reconciliation is canceled when this info is not ready, but for consistency reasons I think it's good to have that log.

I want to remove the TODO before merging.

Reply:
Done.

Reply:
I remember why I did it like that in the first place. Previous `lifecycleresouce` was spamming a lot of logs so I wanted so remove a few of them. But as `WorkerASGName` is available most of the time I don't think it makes much sense.